chainforge/react-server/src/backend/errors.ts

```diff
@@ -0,0 +1,6 @@
+import { ErrorBase } from "./utils";
+
+type ErrorName = 
+    | 'DUPLICATE_VARIABLE_NAME'
+
+export class ProjectError extends ErrorBase<ErrorName> {}
```

Comment:
This should be called something relevant, like `DuplicateVariableNameError`. Furthermore (see below) the `DuplicateVariableNameError` might have a constructor that takes the name of the variable as its sole argument. It then constructs the appropriate error message in the constructor() function.